windows/CodePush.cs

```diff
@@ -0,0 +1,668 @@
+                }
+            }
+
+            private CodePushModule codePush;
+
+            public CodePushNativeModule(ReactContext reactContext, CodePushModule codePush) : base(reactContext)
+            {
+                this.reactContext = reactContext;
+                this.codePush = codePush;
+            }
+
+            public override string Name
+            {
+                get
+                {
+                    return "CodePush";
+                }
+            }
+
+            public override IReadOnlyDictionary<string, object> Constants
+            {
+                get
+                {
+                    Dictionary<string, object> constants = new Dictionary<string, object>();
+                    constants["codePushInstallModeImmediate"] = CodePushInstallMode.IMMEDIATE;
+                    constants["codePushInstallModeOnNextRestart"] = CodePushInstallMode.ON_NEXT_RESTART;
+                    constants["codePushInstallModeOnNextResume"] = CodePushInstallMode.ON_NEXT_RESUME;
+                    constants["codePushUpdateStateRunning"] = CodePushUpdateState.RUNNING;
+                    constants["codePushUpdateStatePending"] = CodePushUpdateState.PENDING;
+                    constants["codePushUpdateStateLatest"] = CodePushUpdateState.LATEST;
+                    return constants;
+                }
+            }
+
+            public override void Initialize()
+            {
+                codePush.InitializeUpdateAfterRestart();
+            }
+
+            private async Task LoadBundle()
+            {
+                // #1) Get the private ReactInstanceManager, which is what includes
+                //     the logic to reload the current React context.
+                FieldInfo info = typeof(ReactPage)
+                    .GetField("_reactInstanceManager", BindingFlags.NonPublic | BindingFlags.Instance);
+
+                ReactInstanceManager reactInstanceManager = (ReactInstanceManager)typeof(ReactPage)
+                    .GetField("_reactInstanceManager", BindingFlags.NonPublic | BindingFlags.Instance)
+                    .GetValue(codePush.mainPage);
+
+                // #2) Update the locally stored JS bundle file path
+                Type reactInstanceManagerType = typeof(ReactInstanceManager);
+                string latestJSBundleFile = await codePush.GetJavaScriptBundleFileAsync(codePush.assetsBundleFileName);
+                reactInstanceManagerType
+                    .GetField("_jsBundleFile", BindingFlags.NonPublic | BindingFlags.Instance)
+                    .SetValue(reactInstanceManager, latestJSBundleFile);
+
+                // #3) Get the context creation method and fire it on the UI thread (which RN enforces)
+                Action recreateReactContextAction = () =>
+                {
+                    reactInstanceManager.RecreateReactContextInBackground();
+                };
+                Context.RunOnDispatcherQueueThread(recreateReactContextAction);
```

Comment:
Instead of creating the Action instance, can't we simply pass a reference to "reactInstanceManager.RecrateReactContextInBackground" to this method?